compiler/passes/buildDefaultFunctions.cpp

```diff
@@ -1592,8 +1596,15 @@ static void buildRecordHashFunction(AggregateType *ct) {
       }
       i++;
     }
-    fn->insertAtTail(new CallExpr(PRIM_RETURN, call));
+
+    if (call) {
+      fn->insertAtTail(new CallExpr(PRIM_RETURN, call));
```

Comment:
What's the situation in which we'd hit `call == NULL`?  When all fields are types and params?

Reply:
(and, assuming so, does your test case cover that code path?  I didn't see a case that looked like it did, but am reading fast).

Reply:
Yeah. One example of that happening is the base `_tuple` type before we specialize it.

Reply:
I'll add a test for that case, thanks!

Reply:
Added in d359a82309f8a404a3e911f2061a6cce376ac153.